src/config.c

```diff
@@ -2426,6 +2445,7 @@ standardConfig configs[] = {
     createStringConfig("aof_rewrite_cpulist", NULL, IMMUTABLE_CONFIG, EMPTY_STRING_IS_NULL, server.aof_rewrite_cpulist, NULL, NULL, NULL),
     createStringConfig("bgsave_cpulist", NULL, IMMUTABLE_CONFIG, EMPTY_STRING_IS_NULL, server.bgsave_cpulist, NULL, NULL, NULL),
     createStringConfig("ignore-warnings", NULL, MODIFIABLE_CONFIG, ALLOW_EMPTY_STRING, server.ignore_warnings, "", NULL, NULL),
+    createStringConfig("proc-title-template", NULL, MODIFIABLE_CONFIG, ALLOW_EMPTY_STRING, server.proc_title_template, CONFIG_DEFAULT_PROC_TITLE_TEMPLATE, isValidProcTitleTemplate, updateProcTitleTemplate),
```

Comment:
Might be a personal preference, but I've never really thought you should use both is_valid_fn() and update_fn(). Since failing update_fn() will also rollback. If we just use update_fn(), we can move all the validation logic into redisSetProcTitle instead of splitting it apart. It also removes the awkward "Should never really happen comment", since the function both validates the input and updates.

Reply:
Good point, should have gotten is_valid_fn out when update_fn got in.